js_modules/dagster-ui/packages/ui-core/src/hooks/useQueryPersistedState.tsx

```diff
@@ -113,9 +114,15 @@ export function useQueryPersistedState<T extends QueryPersistedDataType>(
       // the `replace` so that we surface any unwanted loops during development.
       if (process.env.NODE_ENV !== 'production' || !areQueriesEqual(currentQueryString, next)) {
         currentQueryString = next;
-        history.replace(
-          `${history.location.pathname}?${qs.stringify(next, {arrayFormat: 'brackets'})}`,
-        );
+        if (behavior === 'replace') {
+          history.replace(
+            `${history.location.pathname}?${qs.stringify(next, {arrayFormat: 'brackets'})}`,
+          );
+        } else {
+          history.push(
+            `${history.location.pathname}?${qs.stringify(next, {arrayFormat: 'brackets'})}`,
+          );
```

Comment:
Build the string once prior to the conditionals so that if something in the callsite has to change, we don't accidentally overlook one?